### PR TITLE
fix perf issues

### PR DESCRIPTION
**Pull request by Uzay-G**
Leveraging qdrant index.

Can't really test this locally though without going back to my `local-api` branch, but is straightforward, will see if this fixes the Alan issue.

@spolu good to merge?

### CODE REVIEW

**Review by spolu — CHANGES_REQUESTED** [review]

File: core/src/data_sources/data_source.rs [hunk 1]
```diff
@@ -937,11 +937,11 @@ impl DataSource {
                             let filter = qdrant::Filter {
                                 must: vec![
                                     qdrant::FieldCondition {
-                                        key: "document_id".to_string(),
+                                        key: "document_id_hash".to_string(),
                                         r#match: Some(qdrant::Match {
                                             match_value: Some(
                                                 qdrant::r#match::MatchValue::Keyword(
-                                                    d.document_id.clone(),
+                                                    d.hash.clone(),
```
[the remark this answers is not in the document]
Comment (Uzay-G) [reply core/src/data_sources/data_source.rs]: Good now?

[the remark this answers is not in the document]
Comment (spolu) [reply core/src/data_sources/data_source.rs]: The documentId is an artbitrary string

File: core/src/data_sources/data_source.rs [hunk 2]
```diff
@@ -934,15 +934,18 @@ impl DataSource {
                                 return Ok(d);
                             }
 
+                            let mut hasher = blake3::Hasher::new();
```
Comment (spolu) [inline core/src/data_sources/data_source.rs:937]: For readibility and consistency with the rest of the code base. Lets finalize and assign to `document_id_hash` here

**Review by spolu — CHANGES_REQUESTED** [review]

File: core/src/data_sources/data_source.rs (hunk 2 above)
Comment (Uzay-G) [reply core/src/data_sources/data_source.rs:937]: Okay

File: core/src/data_sources/data_source.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (spolu) [reply core/src/data_sources/data_source.rs]: Not sure I understand?

A fixed length string of predictable format is much easier (and cheaper) to index

**Review by spolu — APPROVED** [review]

File: core/src/data_sources/data_source.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (Uzay-G) [reply core/src/data_sources/data_source.rs]: Ah ok, I understand now.

**Comment by spolu** [comment]
Awesome :+1:

**Pull request merged**